src/extra.i

```diff
@@ -3638,38 +3799,132 @@ static void JM_make_text_block(fz_stext_block *block, PyObject *block_dict, int
+                {
+                    block_dict = PyDict_New();
+                    block_n++;
+                    DICT_SETITEM_DROP(block_dict, dictkey_type, Py_BuildValue("i", block->type));
+                    DICT_SETITEM_DROP(block_dict, dictkey_number, Py_BuildValue("i", block_n));
+                    DICT_SETITEMSTR_DROP(block_dict, "flags", Py_BuildValue("i", block->u.t.flags));
+                    JM_make_text_block(block, block_dict, raw, text_buffer, tp_rect, dev_flags);
+                    LIST_APPEND_DROP(block_list, block_dict);
+                }
+            break;
+
+            case FZ_STEXT_BLOCK_IMAGE:
+                if (fz_contains_rect(tp_rect, block->bbox) || fz_is_infinite_rect(tp_rect))
+                {
+                    block_dict = PyDict_New();
+                    block_n++;
+                    DICT_SETITEM_DROP(block_dict, dictkey_type, Py_BuildValue("i", block->type));
+                    DICT_SETITEM_DROP(block_dict, dictkey_number, Py_BuildValue("i", block_n));
+                    DICT_SETITEM_DROP(block_dict, dictkey_bbox, JM_py_from_rect(block->bbox));
+                    JM_make_image_block(block, block_dict);
+                    LIST_APPEND_DROP(block_list, block_dict);
+                }
+            break;
+
+            case FZ_STEXT_BLOCK_VECTOR:
+                if (JM_rects_overlap(tp_rect, block->bbox) || fz_is_infinite_rect(tp_rect))
+                {
+                    block_dict = PyDict_New();
+                    block_n++;
+                    DICT_SETITEM_DROP(block_dict, dictkey_type, Py_BuildValue("i", block->type));
+                    DICT_SETITEM_DROP(block_dict, dictkey_number, Py_BuildValue("i", block_n));
+                    JM_make_vector_block(block, block_dict);
+                    LIST_APPEND_DROP(block_list, block_dict);
+                }
+            break;
+
+            case FZ_STEXT_BLOCK_GRID:
+                if (JM_rects_overlap(tp_rect, block->bbox) || fz_is_infinite_rect(tp_rect))
+                {
+                    block_dict = PyDict_New();
+                    block_n++;
+                    DICT_SETITEM_DROP(block_dict, dictkey_type, Py_BuildValue("i", block->type));
+                    DICT_SETITEM_DROP(block_dict, dictkey_number, Py_BuildValue("i", block_n));
+                    JM_make_grid_block(block, block_dict);
+                    LIST_APPEND_DROP(block_list, block_dict);
+                }
+            break;
+        }
+        block = block->next;
+    }
+    return block_n;
+}
+
+void _getTableBlocks(fz_stext_page *tp, PyObject *table_dict, PyObject *bbox, int dev_flags)
 {
     fz_context* ctx = mupdf::internal_context_get();
+    fz_rect bounds = JM_rect_from_py(bbox);
     fz_stext_block *block;
     fz_buffer *text_buffer = fz_new_buffer(ctx, 128);
-    PyObject *block_dict, *block_list = PyList_New(0);
+    PyObject *block_list = PyList_New(0);
+    block = fz_find_table_within_bounds(ctx, tp, bounds);
```

Comment:
This looks very unsafe if fz_find_table_within_bounds() raises a mupdf exception - it'll longjmp into nowhere.

At the very least we should call `mupdf::ll_fz_find_table_within_bounds()` so that mupdf exceptions are converted to c++ exceptions. But even then it'll leak resources.